modules/cms/widgets/AssetList.php

```diff
@@ -662,7 +663,9 @@ protected function checkUploadPostback()
             /*
              * Accept the uploaded file
              */
-            $uploadedFile->move($this->getCurrentPath(), $uploadedFile->getClientOriginalName());
+            $uploadedFile = $uploadedFile->move($this->getCurrentPath(), $uploadedFile->getClientOriginalName());
+
+            @chmod($uploadedFile, octdec(Config::get('cms.defaultMask.file')));
```

Comment:
@kmarty did you test this? Shouldn't it be getPath() not the object by itself?

Reply:
Also should there be some sort of check to see if there's actually a valid permission value set in the config since the default is null?

Reply:
> @kmarty did you test this? Shouldn't it be getPath() not the object by itself?
Definitely not `getPath()` in this case.
Frankly, I wasn't sure whether I have to use `getRealPath()` or `__toString()` (probably cleaner(?)), but in real test, simply `$uploadedFile` works too.

Reply:
Let's go with getRealPath() if it works, that'll be more explicit about what's going on.